### DISCUSSION TITLE
[Tool] Converting NextJS Docs to PDF for Researchers & Developers: Streamline AI Model Training

### DISCUSSION THREAD

**Discussion by ala-garbaa-pro** [vercel/next.js#49746] [Ideas]
**Tool Repo:** https://github.com/ala-garbaa-pro/nextjs-md-docs-to-pdf

### Goals

1. Simplify the process of converting NextJS documentation to a PDF file
2. Make documentation easier to read and distribute
3. Help researchers and developers train AI models using documentation

### Non-Goals

1. Implement new features to the NextJS technology itself
2. Change the format or structure of the NextJS documentation

### Background

As a developer, I often find it challenging to navigate through documentation that is not in an easy-to-read format. Additionally, having standardized documentation is helpful when training AI models to understand a specific technology or API. While there are other tools available that can convert documentation to different formats, I found that they were either too complex or lacked the features I needed.

### Proposal

The tool converts NextJS documentation in Markdown format to a PDF file in a standardized format. It fetches the latest version of the NextJS documentation from the official API, clones the documentation repository to a local folder, merges all Markdown files into a single file, and converts the merged Markdown file to PDF format.

I am interested in contributing to the NextJS community and would be happy to discuss further if you have any questions.

**Comment by lewxdev** [comment]
An option for MDX would be nice
